deepforest/scanner.py

```diff
@@ -0,0 +1,58 @@
+"""
+Implement methods on extracting structured data such as images or sequences.
+"""
+import torch
+import torch.nn as nn
+
+
+class ImageScanner(object):
+
+    def __init__(self, kernel_size, stride=1, padding=0, backend="numpy"):
+        self.kernel_size = kernel_size
+        self.stride = stride
+        self.padding = padding
+        self.backend = backend
+
+    def _validate_parameters(self, image_size):
+        pass
+
+    def _torch_transform(self, X):
+        """the shape of X is n, c, h, w"""
+        err_msg = "_torch_transform only accepts torch.Tensor as input, \
+                  but got {}"
+        assert isinstance(X, torch.Tensor), err_msg.format(type(X))
+        if X.dtype != torch.uint8:
+            msg = "Warning! ImageScanner will force to transform data \
+                   type of {} to uint8."
+            print(msg.format(X.dtype))
+            X = X.byte()
```

Comment:
The input to `_torch_transform` is a TensorDataset, which wraps the original dataset in the form of Numpy array. The output of `_torch_transform` is a Numpy array of the size (n_samples * n_patches_per_sample, kernel_size * kernel_size), and the data type is `np.uint8`.